### ISSUE TITLE
sklearn2pmml: pmml version 4.2.1 required

### ISSUE THREAD

**Issue by deepanks** [jpmml/sklearn2pmml#286]
Is there way to generate pmml version 4.2.1 for xgboost classifier model. I am trying using older sklearn2pmml module (0.25.0) but facing errors. Can you specify packages version required to generate the pmml file needed.

Thanks

**Comment by vruusmann** [comment]
All JPMML family conversion libraries are generating PMML documents that conform to the latest PMML schema version. As of today (June 2021), this is [PMML 4.4(.1)](http://dmg.org/pmml/pmml-v4-4-1.html).

> Is there way to generate pmml version 4.2.1 for xgboost classifier model

Take a PMML 4.4 document, and use automatic conversion tools to translate it back to PMML 4.3 or 4.2(.1) document.

> I am trying using older sklearn2pmml module (0.25.0) but facing errors.

Legacy SkLearn2PMML package versions should be pairsed with legacy XGBoost package versions. You should be able to figure out the range of supported XGBoost package versions yourself by keeping track of the enclosed JPMML-XGBoost package versions.

Legacy SkLearn2PMML package versions will surely run into trouble when combined with latest XGBoost package versions. For example, they have no idea of XGBoost JSON representation.

> Can you specify packages version required to generate the pmml file needed.

This has been discussed numerous times in GitHub issues and/or JPMML mailing list. Please use the search functionality.

**Comment by deepanks** [comment]
Sorry, I am actually very new to this. Can you help me on how I can find the correct xgb version
Currently, I have
pandas 1.2.2
sklearn2pmml 0.25.0
xgboost 1.4.2

Currently getting the error
![image](https://user-images.githubusercontent.com/42202902/123318219-9ccf6380-d54c-11eb-9532-24ad0361a0a4.png)
![image](https://user-images.githubusercontent.com/42202902/123318282-ace74300-d54c-11eb-83a1-041acf8abdbe.png)

**Comment by vruusmann** [comment]
> Can you help me on how I can find the correct xgb version
> 
> sklearn2pmml 0.25.0

SkLearn2PMML 0.25.0 is depends on JPMML-SkLearn 1.4.0, which in turn depends on JPMML-XGBoost 1.2.4.

Now, if you check out JPMML-XGBoost version history, then you can see that it has been developed around October 2017 (ie. three and a half years ago!), when XGBoost v0.6 -- v0.7 was all the rage:
https://github.com/dmlc/xgboost/releases/tag/v0.60

TLDR: Looks like you'd need to be combining SkLearn2PMML 0.25.0 with XGBoost 0.60.
